crates/agglayer-aggregator-notifier/src/certifier/mod.rs

```diff
@@ -329,6 +344,57 @@ where
             }
         };
 
+        let aggchain_vkey = match certificate.aggchain_data {
+            AggchainData::ECDSA { .. } => None,
+            AggchainData::Generic { ref proof, .. } => {
+                if certificate.custom_chain_data.len() < 2 {
+                    return Err(CertificationError::Types {
+                        source: agglayer_types::Error::InvalidCustomChainDataLength {
+                            expected_at_least: 2,
+                            actual: certificate.custom_chain_data.len(),
+                        },
+                    });
+                }
+
+                let aggchain_vkey_selector: u16 = {
+                    let mut selector = [0u8; 2];
+                    selector.copy_from_slice(&certificate.custom_chain_data[0..2]);
```

Comment:
Minor: can use `slice::get` to roll the length check and slicing into one, avoiding the panicking slicing operation:
```suggestion
                let aggchain_vkey_selector: u16 = {
                    let selector_slice = certificate
                        .custom_chain_data
                        .get(0..2)
                        .ok_or(CertificationError::Types {
                            source: agglayer_types::Error::InvalidCustomChainDataLength {
                                expected_at_least: 2,
                                actual: certificate.custom_chain_data.len(),
                            },
                        })?;
                        
                    let mut selector = [0u8; 2];
                    selector.copy_from_slice(selector_slice);
```